src/python/CMSSpark/condor_crab_unique_users.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Author: Christian Ariza <christian.ariza AT gmail [DOT] com>
+"""
+Generate datasets and plots for CRAB unique users count
+either by week of year or by month for a given calendar year.
+"""
+import os
+from datetime import datetime, timezone, timedelta
+from pyspark import SparkContext
+from pyspark.sql import SparkSession
+from pyspark.sql.types import (
+    StructType,
+    LongType,
+    StringType,
+    StructField,
+    DoubleType,
+)
+from pyspark.sql.functions import (
+    regexp_extract,
+    date_format,
+    from_unixtime,
+    substring_index,
+    col,
+    input_file_name,
+    concat,
+    year,
+    month,
+    weekofyear,
+    lpad,
+    countDistinct,
+)
+import matplotlib.pyplot as plt
+import seaborn as sns
```

Comment:
Do we really need two plotting libraries? Can we use just one? It will reduce number of dependencies.

Reply:
We could use just `matplotlib`, seaborn is just a wrapper on top of it with more sensible defaults and some predefined plots (that in this case, we don't need), however as seaborn is a dependency for other scripts (so it is already a dependency of the module) so it will not imply a new dependency and using it here result in nicer plot that the one with only matplotlib. Seaborn is already available in the LCG_96python3 environment.

Reply:
ahh, ok, then